### PR TITLE
Adding bulk actions to 1.1

### PR DESCRIPTION
**Pull request by ssanjay1**
We added bulk actions to the top level client ontology due to difficulties wrangling with the types that would conditionally enforce options and return types based on what parameters were passed in (array vs single input)

### CODE REVIEW

File: packages/legacy-client/src/client/baseTypes/ActionType.ts
Comment (ssanjay1) [inline packages/legacy-client/src/client/baseTypes/ActionType.ts:166]: Didn't touch this, but for some reason had to add this type, or else checks error out with something along the lines of :
`Argument of type '{ apiName: string; primaryKey: any; get: () => Promise<Result<OntologyObject<string, NonNullable<ParameterValue>>, GetObjectError>>; }' is not assignable to parameter of type 'never'.ts(2345)`

File: packages/legacy-client/src/client/ontology.ts
Comment (ssanjay1) [inline packages/legacy-client/src/client/ontology.ts:44]: I only added this here, and not in the actual OntologyDefinition since this seems like we're only adding this for 1.1

File: packages/legacy-client/src/client/actions/actions.ts
Comment (ssanjay1) [inline packages/legacy-client/src/client/actions/actions.ts:191]: Right now I'm just following pattern of how we've handled other actions with no parameters, but the way its setup bulk actions don't really work for actions without parameters. This _could_ be modified so it accepts an array of `{}` potentially? And we just use those to tell us how many empty requests we need for the batch endpoint?

File: packages/legacy-client/src/client/baseTypes/ActionType.ts
Comment (ssanjay1) [inline packages/legacy-client/src/client/baseTypes/ActionType.ts:37]: Don't want users trying to pass in any sort of execution mode as that isn't supported by the endpoint

**Review by ssanjay1 — COMMENTED** [review]
Left some comments as well, also general question, where is the best place to expose some sort of docs to indicate that the bulk actions endpoint only supports up to 20 requests at a time?

File: packages/generator/src/v1.1/generateBulkActions.ts
Comment (ericanderson) [inline packages/generator/src/v1.1/generateBulkActions.ts:36]: I think `.addedObjects`/`.modifiedObjects` is a `Set<string>`. So we are converting it to an array so we can `forEach` add it to a different set. At the very least you could just `forEach` on the returned sets instead of the memory copy.

File: packages/legacy-client/src/client/actions/actions.ts
Comment (ericanderson) [reply packages/legacy-client/src/client/actions/actions.ts:191]: Yeah, I think `{}[]` or `undefined[]` is better than nothing, otherwise you arent really calling bulk.

File: packages/legacy-client/src/client/actions/createActionProxy.ts
Comment (ericanderson) [inline packages/legacy-client/src/client/actions/createActionProxy.ts:53]: nit: unnecessary change

File: packages/legacy-client/src/client/actions/createActionProxy.ts
Comment (ericanderson) [inline packages/legacy-client/src/client/actions/createActionProxy.ts:57]: nit: unnecessary change

File: packages/legacy-client/src/client/baseTypes/ActionType.ts
Comment (ericanderson) [reply packages/legacy-client/src/client/baseTypes/ActionType.ts:37]: The problem with this though is it is suggested in intellisense even though it can only be undefined.

It can just be omitted right? https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcBCBXANgawIIDGMAlhAHYCiIwBGJ5A8mPWQM5wC8cA3gFBxwAthAAmwAPwAuOGWAA3YFADc-OFGAwMUSiOIxWUmRkEAjRXF4BfXmIJYAhurgAzDGSKkyLsgAoIYaUxcQhYqGjpPJhZWAEppOQhiERVeZ18+AQEAeiy4YjIYYCwsYlZgNlRWAAsIAHd2AANhMQarGN5eUEhYBGRUIPwPcjDaFijPVgAmTh5VdU1tCl19QzJjMygLa1sHJ1d3Fm9JvwD0bEHQ6lHI5gnJuLgEpJU044zM5uBpAFYlOByZBB4FhgABzexYOBgRz2QR5ZxwRAQDBwABWGFY8Aggj0bSAA

File: packages/legacy-client/src/client/baseTypes/ActionType.ts
Comment (ericanderson) [reply packages/legacy-client/src/client/baseTypes/ActionType.ts:166]: Hmm. I am a little worried this means we broke something somewhere. If we need to keep a change like this we could at least type it to the real shape we are pushing to prevent future errors.

File: packages/legacy-client/src/client/baseTypes/ActionType.ts
Comment (ericanderson) [inline packages/legacy-client/src/client/baseTypes/ActionType.ts:218]: same here

File: packages/legacy-client/src/client/baseTypes/ActionType.ts
Comment (ericanderson) [inline packages/legacy-client/src/client/baseTypes/ActionType.ts:219]: This seems like a copy/paste of the above.  Maybe extract function for getting the `edits` shape (`{type:"edits", added,modified}`)?

File: packages/legacy-client/src/client/net/executeAction.ts
Comment (ericanderson) [inline packages/legacy-client/src/client/net/executeAction.ts:99]: Given empty array is used here, does this mean that any actions that take zero arguments can never be called in bulk? If thats the intention, should we just exclude them in the type mapping for creating bulk actions?

**Review by styu — COMMENTED** [review]
defer to @ericanderson on the meat of the changes

File: packages/legacy-client/src/client/ontology.ts
Comment (ericanderson) [reply packages/legacy-client/src/client/ontology.ts:44]: That would be correct, since there is no custom definition for actions vs bulk actions. Only the runtime `client.ontology.` should be showing `bulkActions`.

**Review by ericanderson — COMMENTED** [review]
Some notes to fix up before merging.

File: packages/legacy-client/src/client/baseTypes/ActionType.ts
Comment (ssanjay1) [reply packages/legacy-client/src/client/baseTypes/ActionType.ts:37]: Ah yea you're right, I think I needed this before when we tried the original syntax and I had some conditional types for the options, but yea I think I can just omit now, on it

File: packages/legacy-client/src/client/actions/actions.ts
Comment (ssanjay1) [reply packages/legacy-client/src/client/actions/actions.ts:191]: Would `undefined[]` be preferable here? Because with `{}[]`won't that just let you pass in any sort of object https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcBCBXANgawIIDGMAlhAHYCiIwBGJ5A8mPWQM5wC8cA3gFBxwAthAAmwAPwAuOGQyCARsCgBufnCjAYGKJRHEYrKTLmKocXgF9eYglgCGGuADMMZIqTLOyACghhpmLiELFQ0dB5MLKwAlNIAbhDEIqq8Tj7c5gICAPTZcMRkMMBYWMSswGyorAAWEADu7AAGwmKNltG8vKCQsAjIqIH47uShtCyRHqwATJw8ahpaOhR6BkayCkrmVjb2ji5uLF5TADwACnCgRWQi7IPBHqPhjMyTUwB8vv5wp7FwCUmqNJTbx8LJCUTAaQAVlyMgg8CwwAA5nYsHAwA47IJ8k44IgIBg4AArDCseAQQT6dqdIHHO7DSjUMYRF7kVgfUFZFqQuAwvJkeFwREotEYqBYnF4gnE0nkykwaldcDQeBIFBweTYHDAQTMRCzTkFapKSHcCwAbQAuqorLwCGyEZoyVBEJJNbgdXqDUaTZJzdwAJIEOzXerVOxFOJKSQARgsABoeBZLRYgA

Comment (ssanjay1) [reply packages/legacy-client/src/client/actions/actions.ts:191]: But also feel like it could be weird synatx to do something like `client.ontology.bulkActions.noParamaction([undefined, undefined,...], {options})`?

Comment (ssanjay1) [reply packages/legacy-client/src/client/actions/actions.ts:191]: Oh wait could we use `Record<string,never>[]`?

File: packages/generator/src/v1.1/generateBulkActions.ts
Comment (ssanjay1) [reply packages/generator/src/v1.1/generateBulkActions.ts:36]: Ah yea I just pulled this from the `generateActions`, it looks like we converted to Array so we could just do a simple join on line 74 when generating the edits return type

**Review by ericanderson — APPROVED** [review]

**Pull request merged**
